src/features/TodoGenerator.js

```diff
@@ -1,13 +1,17 @@
 import React, { useState } from "react";
-
+import { useDispatch } from "react-redux";
+import { addTodoItem } from "./TodoSlice.js";
 export default function TodoGenerator(props) {
   const [todo, setTodo] = useState("");
   const onInputChange = (event) => {
     setTodo(event.target.value);
   };
+  const dispatch = useDispatch();
+
   const onAdd = () => {
-    props.addTodo(todo);
     setTodo("");
+    dispatch(addTodoItem(todo));
```

Comment:
`dispatch(addTodoItem(todo))` may come before `setTodo("")` to avoid dispatching empty string